mysensors/mysensors.py

```diff
@@ -114,7 +114,9 @@ def _handle_heartbeat(self, msg):
 
     def _handle_internal(self, msg):
         """Process an internal protocol message."""
-        if msg.sub_type == self.const.Internal.I_ID_REQUEST:
+        if msg.sub_type == self.const.Internal.I_VERSION:
```

Comment:
You have to overwrite this method in the child class TCPGateway and add the logic there. Make sure you also call this parent method via `super` and handle the return value.